### DISCUSSION TITLE
[@types/ws] Type 'Server' is not generic

### DISCUSSION THREAD

**Discussion by maiconjobim** [DefinitelyTyped/DefinitelyTyped#67393] [Issues with a @types package]
When I compile from typescript I get this bug

  "@types/ws": "^8.5.7"
 "ws": "^8.14.2"

`node_modules/@types/ws/index.d.ts:321:18 - error TS2315: Type 'Server' is not generic.

321         server?: HTTPServer<V> | HTTPSServer<V> | undefined;`

[a comment by an automated account is not shown]

**Comment by k-yle** [comment]
This was caused by #65501. #65849 proposed reverting this change, but that PR was abandoned

**Comment by unphased** [comment]
The error I was having looks like 

    Type 'Server<typeof WebSocket, typeof IncomingMessage>' is missing the following properties from type 'WebSocket': binaryType, bufferedAmount, extensions ... 

Is it the same issue? I'm going to try to get by without using @types/ws to use ws from my typescript code for now.

**Reply by k-yle** [reply]
Yes that's the same issue. I suggest joining the discussion on #65501 if this is still causing problems for you.

**Reply by unphased** [reply]
It's not really possible to understand that thread. Mostly people talking about angular, which I don't care about. Just using the ws lib here to make a barebones ws server running on node. Something is extremely broken with the types and it looks like it's been that way for 6 months. The guidance that will probably work is to pin the version to before the generics change. I will try that but would love to know who that change benefits.
